### PR TITLE
Allow usage of $HOME/.ssh/config for each user

### PR DESCRIPTION
**Pull request by jabley**
My new GDS machine has a username of jamesabley. The initial machine
that I used at work was my personal one (used for my first 5 months) and
had a username of jabley. My first GDS machine also had a username of
jabley.

This change to the fabfile means that the entry in my $HOME/.ssh/config
file for `User` [1] is honoured, and I can do:

```fab production vm.disk -H some.host```

rather than:

```fab -u jabley production vm.disk -H some.host```

[1] https://github.com/alphagov/gds-boxen/blob/d25c24eec813f98653a60d7e2b57d48173f4d454/modules/people/templates/jabley/ssh_config#L52-L56

### CODE REVIEW

**Comment by bradleywright** [comment]
The issue with the default username can also be resolved in `~/.fabricrc`:

```
user = username
```

Not sure how others feel about this.

**Comment by evilstreak** [comment]
I think it already uses `$HOME/.ssh/config`, as my local and remote usernames don't match and I have this in my config:

```
Host *.production
  User dominicbaggott
  ProxyCommand …

Host *.staging
  User dominicbaggott
  ProxyCommand …

Host *.preview
  User dominicbaggott
  ProxyCommand …
```

**Comment by mattbostock** [comment]
I like the idea in principle.

One possible side-effect of this is we might see non-deterministic behaviour as everyone's SSH configuration will vary (a concern since we use these scripts when responding to incidents).

But knowing my own configuration, I'd be happy with this :+1:

**Comment by ajlanghorn** [comment]
@surminus and I looked at this, but since we can't test it (I'm `ajlanghorn` everywhere now, and @surminus will be `paulmartin` everywhere since our SSH username policy was changed a while back), I'm not :+1: with merging this, given @mattbostock's comment above.

TL;DR: This might break things for others. Not happy merging this personally.

**Comment by ajlanghorn** [comment]
@jabley Is there a reason you _don't_ want to use `~/.fabricrc` and prefer this method? Is it just simplicity  inside your local tilde dir?

**Comment by jabley** [comment]
https://github.com/alphagov/gds-boxen/pull/172

I prefer to scope within `~/.ssh/config`, rather than adding another `rc` file to the
mix. I can live with it though.
